renku_notebooks/api/classes/server.py

```diff
@@ -0,0 +1,399 @@
+import os
+import requests
+from kubernetes import client
+from kubernetes.client.rest import ApiException
+from hashlib import md5
+import base64
+import json
+from urllib.parse import urlparse
+from uuid import uuid4
+from flask import make_response, jsonify
+import logging
+from urllib.parse import urljoin
+
+
+from ...util.check_image import parse_image_name, get_docker_token, image_exists
+from ...util.kubernetes_ import (
+    get_k8s_client,
+    secret_exists,
+    filter_pods_by_annotations,
+)
+
+
+class UserServer:
+    """Represents a jupuyterhub session."""
+
+    def __init__(
+        self,
+        user,
+        namespace,
+        project,
+        branch,
+        commit_sha,
+        notebook,
+        image,
+        server_options,
+    ):
+        self._renku_annotation_prefix = "renku.io/"
+        self._get_environment_vars()
+        self._user = user
+        self._k8s_client, self._k8s_namespace = get_k8s_client()
+        self.safe_username = self._user.safe_username
+        self.namespace = namespace
+        self.project = project
+        self.branch = branch
+        self.commit_sha = commit_sha
+        self.notebook = notebook
+        self.image = image
+        self.server_options = server_options
+        self.server_name = self.make_server_name(
+            self.namespace, self.project, self.branch, self.commit_sha
+        )
+
+    def _get_environment_vars(self):
```

Comment:
@ableuler I agree. I renamed and kept a small portion of this method that now only checks that the flask app config contains the necessary things. Also since both the server and user classes are so tightly coupled to jupyterhub I did not pass the config in the constructor but rather used the `current_app.config` object. However if you think that I should pass the config in the constructor to both classes I can easily make that change.

Reply:
ok, it's probably the more pragmatic approach.